File: php/features.php
<?php
declare(strict_types=1);

// Radiorecord SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class RadiorecordFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new RadiorecordBaseFeature();
            case "test":
                return new RadiorecordTestFeature();
            default:
                return new RadiorecordBaseFeature();
        }
    }
}

File: py/features.py
# Radiorecord SDK feature factory

from feature.base_feature import RadiorecordBaseFeature
from feature.test_feature import RadiorecordTestFeature


def _make_feature(name):
    features = {
        "base": lambda: RadiorecordBaseFeature(),
        "test": lambda: RadiorecordTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
